### PR TITLE
[opt](load) check -238 segment number limit earlier

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Previously we check segment number when build rowset because we don't want throw an error in the intermediate state, hoping segment compaction can reduce segment number later.

This PR adds a earlier segment number check, while doing segment compaction.
Immediately throw an error if `_segcompacted_point` is greater than the segment number limit.

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -514,26 +514,27 @@ Status BetaRowsetWriter::_rename_compacted_indices(int64_t begin, int64_t end, u
...
+    if (_is_doing_segcompaction.exchange(true)) {
         return status;
     }
     if (_segcompaction_status.load() != OK) {
         status = Status::Error<SEGCOMPACTION_FAILED>(
                 "BetaRowsetWriter::_segcompaction_if_necessary meet invalid state, error code: {}",
                 _segcompaction_status.load());
-    } else if ((_num_segment - _segcompacted_point) >= config::segcompaction_batch_size) {
+    } else {
+        status = _check_segment_number_limit(_segcompacted_point);
```
Comment (liaoxin01) [inline be/src/olap/rowset/beta_rowset_writer.cpp:535]: Why use _segcompacted_point instead of _num_segcompacted ?

**Comment by kaijchen** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Comment by liaoxin01** [comment]
run buildall

**Review by sollhui — APPROVED** [review]
LGTM

**Pull request merged**